broker/setup_broker/lsst/setup_broker.sh

```diff
@@ -0,0 +1,146 @@
+
+# "False" uses production resources
+# any other string will be appended to the names of all resources
+testid="${1:-test}"
+# "True" tearsdown/deletes resources, else setup
+teardown="${2:-False}"
+# name of the survey this broker instance will ingest
+survey="${3:-lsst}"
+schema_version="${4:-7.3}"
+versiontag=v$(echo "${schema_version}" | tr . _) # 7.3 -> v7_3
+region="${5:-us-central1}"
+zone="${region}-a"  # just use zone "a" instead of adding another script arg
+
+PROJECT_ID=$GOOGLE_CLOUD_PROJECT # get the environment variable
+
+#--- Make the user confirm the settings
+echo
+echo "setup_broker.sh will run with the following configs: "
+echo
+echo "GOOGLE_CLOUD_PROJECT = ${PROJECT_ID}"
+echo "survey = ${survey}"
+echo "testid = ${testid}"
+echo "schema_version = ${schema_version}"
+echo "teardown = ${teardown}"
+echo
+echo "Continue?  [y/(n)]: "
+
+read -r continue_with_setup
+continue_with_setup="${continue_with_setup:-n}"
+if [ "$continue_with_setup" != "y" ]; then
+    echo "Exiting setup."
+    echo
+    exit
+fi
+
+# function used to define GCP resources; appends testid if needed
+define_GCP_resources() {
+    local base_name="$1"
+    local testid_suffix=""
+
+    if [ "$testid" != "False" ]; then
+        if [ "$base_name" = "$survey" ]; then
+            testid_suffix="_${testid}"  # complies with BigQuery naming conventions
+        else
+            testid_suffix="-${testid}"
+        fi
+    fi
+
+    echo "${base_name}${testid_suffix}"
+}
+
+#--- GCP resources used directly in this script
+broker_bucket=$(define_GCP_resources "${PROJECT_ID}-${survey}-broker_files")
+topic_alerts_raw=$(define_GCP_resources "${survey}-alerts_raw")
+subscription_alerts_raw=$(define_GCP_resources "${survey}-alerts_raw-counter")
+topic_alerts=$(define_GCP_resources "${survey}-alerts")
+subscription_alerts=$(define_GCP_resources "${survey}-alerts-counter")
+subscription_reservoir=$(define_GCP_resources "${survey}-alerts-reservoir")
+
+# function used to create (or delete) GCP resources
+manage_resources() {
+    local mode="$1"  # setup or teardown
+    local environment_type="production"
+
+    if [ "$testid" != "False" ]; then
+        environment_type="testing"
+    fi
+
+    if [ "$mode" = "setup" ]; then
+        # create broker bucket and upload files
+        echo
+        echo "Creating broker_bucket and uploading files..."
+        gsutil mb -b on -l "${region}" "gs://${broker_bucket}"
+        ./upload_broker_bucket.sh "${broker_bucket}"
+
+        # create a firewall rule to open the port used by Kafka/Rubin LSST
+        # on any instance with the flag --tags=tcpport9094
+        echo
+        echo "Configuring Rubin/Kafka firewall rule..."
+        firewallrule="tcpport9094"
+        gcloud compute firewall-rules create "${firewallrule}" \
+            --allow=tcp:9094 \
+            --description="Allow incoming traffic on TCP port 9094" \
+            --direction=INGRESS \
+            --enable-logging
```

Comment:
Is this necessary for LSST?

Reply:
I'm not entirely sure. The firewall rule allows incoming traffic on TCP port 9094. If this code block is removed, I imagine that incoming traffic is now prevented from reaching the port(?)

Reply:
Sorry, I think I was conflating the firewall and the authentication mechanisms in my head when I wrote this.